presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotAggregationProjectConverter.java

```diff
@@ -217,17 +231,45 @@ private PinotExpression handleFunction(
             CallExpression function,
             Map<VariableReferenceExpression, PinotQueryGeneratorContext.Selection> context)
     {
-        switch (function.getDisplayName().toLowerCase(ENGLISH)) {
+        String functionName = function.getDisplayName().toLowerCase(ENGLISH);
+        switch (functionName) {
             case "date_trunc":
                 boolean useDateTruncation = PinotSessionProperties.isUseDateTruncation(session);
                 return useDateTruncation ?
                         handleDateTruncationViaDateTruncation(function, context) :
                         handleDateTruncationViaDateTimeConvert(function, context);
+            case "array_max":
+            case "array_min":
+                return derived(String.format("%s(%s)",
+                        ARRAY_AGGREGATION_ALLOWED_PUSHDOWN_MAP.get(functionName),
```

Comment:
Should we require that the .get does not return a null here ?

Reply:
done.